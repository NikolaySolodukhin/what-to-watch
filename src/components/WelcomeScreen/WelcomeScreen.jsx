import React, {Fragment, PureComponent} from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import CatalogMoviesList from './../CatalogMovieList/CatalogMovieList.jsx';
import GenresList from './../GenreList/GenreList.jsx';
import Footer from './../../components/Footer/Footer.jsx';
import Header from './../../components/Header/Header.jsx';
import MovieCardInfo from './../../components/MovieCardInfo/MovieCardInfo.jsx';
import withActiveFilm from '../hocs/WithActiveFilm/WithActiveFilm.jsx';
import {getGenres} from './../../reducer/films/selectors';
import {getActiveFilms, getActiveGenre, getPromoFilm} from './../../reducer/films/selectors';
import {ActionCreator as ActionCatalog} from './../../reducer/films/films';

const CatalogMoviesListWrap = withActiveFilm(CatalogMoviesList);

class WelcomeScreen extends PureComponent {
  render() {
    const {activeGenre, genres, setActiveGenre, filmsList, promoFilm} = this.props;
    return (
      <Fragment>
        <section className="movie-card">
          <div className="movie-card__bg">
            <img src={promoFilm && promoFilm.backgroundImage} alt={promoFilm && promoFilm.name} />
          </div>

          <h1 className="visually-hidden">WTW</h1>

          <Header withAuth/>
          {promoFilm && <MovieCardInfo film={promoFilm} withPoser/>}
        </section>

        <div className="page-content">
          <section className="catalog">
            <GenresList
              genres={genres}
              activeGenre={activeGenre}
              onGenreLinkClick={setActiveGenre} />
            <CatalogMoviesListWrap filmsList={filmsList} />
          </section>
          <Footer/>
        </div>
      </Fragment>
    );
  }
}

WelcomeScreen.propTypes = {
  filmsList: PropTypes.arrayOf(PropTypes.object).isRequired,
  promoFilm: PropTypes.object,
  genres: PropTypes.array.isRequired,
  activeGenre: PropTypes.string.isRequired,
  setActiveGenre: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => ({
  ...ownProps,
  filmsList: getActiveFilms(state),
  activeGenre: getActiveGenre(state),
  genres: getGenres(state),
  promoFilm: getPromoFilm(state),
});

const mapDispatchToProps = (dispatch) => ({
  setActiveGenre: (genre) => dispatch(ActionCatalog.setActiveGenre(genre)),
});

export {WelcomeScreen};

export default connect(mapStateToProps, mapDispatchToProps)(WelcomeScreen);
