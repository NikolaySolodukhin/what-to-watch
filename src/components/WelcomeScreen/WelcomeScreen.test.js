import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from 'enzyme-to-json';
Enzyme.configure({adapter: new Adapter()});

import {WelcomeScreen} from './WelcomeScreen';

describe(`<WelcomeScreen/>`, () => {
  it(`WelcomeScreen renders correctly`, () => {
    const mockProps = {
      filmsList: [{
        id: 1,
        title: `Fantastic Beasts`,
        genreId: 1,
        image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
        video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
      }],
      activeGenre: `Comedies`,
      genres: [`Comedies`],
      setActiveGenre: jest.fn(),
    };
    const tree = shallow(<WelcomeScreen filmsList={mockProps.filmsList}
      genres={mockProps.genres}
      activeGenre={mockProps.activeGenre}
      setActiveGenre={mockProps.setActiveGenre} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
