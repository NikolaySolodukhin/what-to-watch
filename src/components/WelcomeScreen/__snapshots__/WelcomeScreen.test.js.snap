// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeScreen/> WelcomeScreen renders correctly 1`] = `
<Fragment>
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <ConnectFunction
      withAuth={true}
    />
  </section>
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <ConnectFunction
        activeGenre="Comedies"
        genres={
          Array [
            "Comedies",
          ]
        }
        onGenreLinkClick={[MockFunction]}
      />
      <WithActiveFilm
        filmsList={
          Array [
            Object {
              "genreId": 1,
              "id": 1,
              "image": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
              "title": "Fantastic Beasts",
              "video": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
            },
          ]
        }
      />
    </section>
    <Footer />
  </div>
</Fragment>
`;
